File: Resources/TOI-Zero_68/A3/012/012.py
n, x = map(int, input().split())
a = [0] * (n + 1)
for i in range(1, n + 1):
    a[i] = int(input())

visited = [0] * (n + 1)
count = 0

while x > 0 and not visited[x]:
    visited[x] = 1
    count += 1
    x = a[x]

print(count) 

File: Resources/Projects/Archive/2023/Competition/Programming/Contest/TOI-Zero_68/A2/019/TOI-Zero_68/A3/012/012.cpp
#include<bits/stdc++.h>
using namespace std;

#define ll long long

int main(){
	ios::sync_with_stdio(0);
	cin.tie(0);
	int n,x;
    cin >> n >> x;
    int a[n+1];
    for(int i=1;i<=n;i++){
    	cin >> a[i];
	}
    int count=0,visited[n+1];
    memset(visited,0,sizeof(visited));
    while(x>0 && !visited[x]){
    	visited[x]=1;
		count++;
		x=a[x];
	}
    cout << count;
}
